pkg/engine/engine.go

```diff
@@ -955,6 +1042,53 @@ func (e *Engine) verificationOverlapWorker(ctx context.Context) {
 						val = res.Raw
 					}
 
+					// Only check the cache for non-targeted scans.
+					// Targeted scans should always be verified.
+					if e.enableDetectionCache && !isTargetedScan {
+						// Check the cache to see if we've already seen this secret.
+						keyBytes := make([]byte, len(res.Raw)+len(res.RawV2))
+						copy(keyBytes, res.Raw)
+						copy(keyBytes[len(res.Raw):], res.RawV2)
+						keyHash, err := hasher.Hash(keyBytes)
+						if err != nil {
+							ctx.Logger().Error(err, "failed to hash secret")
+							continue
+						}
+
+						cacheVal, exists := e.detectionCache.Get(string(keyHash))
+						if exists {
+							// Update the "res" object with the cached result.
+							res.Verified = cacheVal.Verified
+							res.DecoderType = cacheVal.DecoderType
+							res.DetectorType = cacheVal.DetectorType
+							res.ExtraData = cacheVal.ExtraData
+							res.Redacted = cacheVal.Redacted
+							res.StructuredData = cacheVal.StructuredData
+							res.AnalysisInfo = cacheVal.AnalysisInfo
```

Comment:
This seems error prone. What about storing the `detector.Result` directly instead of a new type?

Reply:
We could, we'd just have to ensure we remove the Raw and RawV2 fields.